plugins/plugin-shiki/src/node/markdown/preWrapperPlugin.ts

```diff
@@ -23,7 +23,7 @@ export const preWrapperPlugin = (
 
     let result = rawFence(...args)
 
-    if (!preWrapper) {
+    if (!preWrapper || !result.startsWith('<pre')) {
       // remove `<code>` attributes
```

Comment:
Need some detailed code about what it's for.

Technically speaking, we need a note describe what  attributes are shiki adding and why should we remove it.

Also I think the logic is still incorrect here.